helpers/announce.js

```diff
@@ -25,8 +26,16 @@ export function announce(message) {
 	treat the change as an update. Firefox sometimes ignores changes if the region
 	and update are made too quickly in succession. RequestAnimationFrame is not
 	sufficient here. */
+	let child;
+	if (Object.keys(message).length === 0 || !Object.keys(messages).includes(message)) {
+		child = document.createTextNode(message);
+		messages[message] = child;
+	} else {
+		child = messages[message];
+		child.parentNode.removeChild(child);
+	}
```

Comment:
Another option here instead of keeping a Map/reference to all the DOM nodes (which can become stale and create circular references between the DOM and JavaScript) would be to do a runtime search for it. Something like this:

```javascript
const elem = [...container.children].find((c) => c.textContent === message);
if (elem) elem.parentNode.removeChild(elem);
container.appendChild(document.createTextNode(message);
```

Reply:
Good idea - that looks a lot cleaner